File: InternMobileApp/src/components/MaintenanceGuard.js
import React, { useState, useEffect } from 'react';
import { View, Text, ActivityIndicator, StyleSheet, Image } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { supabase } from '../lib/supabase';
import { colors } from '../theme/colors';

const MaintenanceGuard = ({ children }) => {
  const [maintenance, setMaintenance] = useState({ loading: true, enabled: false, title: '', message: '' });

  useEffect(() => {
    const checkMaintenance = async () => {
      try {
        const { data, error } = await supabase
          .from('system_config')
          .select('value')
          .eq('key', 'maintenance_mode')
          .maybeSingle();

        if (error) {
          if (error.code !== 'PGRST116') {
             console.error('Error checking maintenance state:', error);
          }
          setMaintenance({ loading: false, enabled: false });
        } else if (data && data.value) {
          setMaintenance({
            loading: false,
            enabled: data.value.enabled,
            title: data.value.title || 'Scheduled Maintenance',
            message: data.value.message || 'We are currently undergoing scheduled maintenance. Please check back soon.'
          });
        }
      } catch (err) {
        console.error('Failed to fetch maintenance status:', err);
        setMaintenance({ loading: false, enabled: false });
      }
    };

    checkMaintenance();
  }, []);

  if (maintenance.loading) {
    return (
      <View style={styles.loadingContainer}>
        <ActivityIndicator size="large" color={colors.primary} />
      </View>
    );
  }

  if (maintenance.enabled) {
    return (
      <View style={styles.container}>
        <Ionicons name="construct" size={64} color={colors.primary} style={styles.icon} />
        <Text style={styles.title}>{maintenance.title}</Text>
        <Text style={styles.message}>{maintenance.message}</Text>
      </View>
    );
  }

  return children;
};

const styles = StyleSheet.create({
  loadingContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.background,
  },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.background,
    padding: 24,
  },
  icon: {
    marginBottom: 24,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: colors.text.primary,
    marginBottom: 16,
    textAlign: 'center',
  },
  message: {
    fontSize: 16,
    color: colors.text.secondary,
    textAlign: 'center',
    lineHeight: 24,
  },
});

export default MaintenanceGuard;

File: intern_mobile_app_flutter/lib/components/maintenance_guard.dart
import 'package:flutter/material.dart';
import '../core/supabase_client.dart';
import '../theme/colors.dart';

class MaintenanceGuard extends StatefulWidget {
  final Widget child;

  const MaintenanceGuard({super.key, required this.child});

  @override
  State<MaintenanceGuard> createState() => _MaintenanceGuardState();
}

class _MaintenanceGuardState extends State<MaintenanceGuard> {
  bool _loading = true;
  bool _enabled = false;
  String _title = '';
  String _message = '';

  @override
  void initState() {
    super.initState();
    _checkMaintenance();
  }

  Future<void> _checkMaintenance() async {
    try {
      final response = await SupabaseClientConfig.client
          .from('system_config')
          .select('value')
          .eq('key', 'maintenance_mode')
          .maybeSingle();

      if (response != null && response['value'] != null) {
        final val = response['value'] as Map<String, dynamic>;
        setState(() {
          _enabled = val['enabled'] == true;
          _title = val['title'] ?? 'Scheduled Maintenance';
          _message = val['message'] ?? 'We are currently undergoing scheduled maintenance. Please check back soon.';
          _loading = false;
        });
        return;
      }
    } catch (e) {
      debugPrint('Error checking maintenance state: $e');
    }

    if (mounted) {
      setState(() {
        _loading = false;
        _enabled = false;
      });
    }
  }

  @override
  Widget build(BuildContext context) {
    if (_loading) {
      return const Scaffold(
        backgroundColor: AppColors.background,
        body: Center(child: CircularProgressIndicator(color: AppColors.primary)),
      );
    }

    if (_enabled) {
      return Scaffold(
        backgroundColor: AppColors.background,
        body: Center(
          child: Padding(
            padding: const EdgeInsets.all(24.0),
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                const Icon(Icons.build, size: 64, color: AppColors.primary),
                const SizedBox(height: 24),
                Text(
                  _title,
                  style: const TextStyle(fontSize: 24, fontWeight: FontWeight.bold, color: AppColors.text),
                  textAlign: TextAlign.center,
                ),
                const SizedBox(height: 16),
                Text(
                  _message,
                  style: const TextStyle(fontSize: 16, color: Color(0xFF94a3b8), height: 1.5),
                  textAlign: TextAlign.center,
                ),
              ],
            ),
          ),
        ),
      );
    }

    return widget.child;
  }
}
